vortex-sampling-compressor/src/compressors/for.rs

```diff
@@ -52,16 +52,30 @@ impl EncodingCompressor for FoRCompressor {
         like: Option<CompressionTree<'a>>,
         ctx: SamplingCompressor<'a>,
     ) -> VortexResult<CompressedArray<'a>> {
-        let (child, min, shift) = for_compress(&PrimitiveArray::try_from(array)?)?;
+        let (child, for_params) = for_compress_parts(&array.clone().into_primitive()?)?;
 
-        let compressed_child = ctx
-            .named("for")
-            .excluding(self)
-            .compress(&child, like.as_ref().and_then(|l| l.child(0)))?;
-        Ok(CompressedArray::new(
-            FoRArray::try_new(compressed_child.array, min, shift).map(|a| a.into_array())?,
-            Some(CompressionTree::new(self, vec![compressed_child.path])),
-        ))
+        for_params
+            .map(|(min, shift)| {
```

Comment:
wdyt about using a match statement here instead of a `map/unwrap_or_else` chain, I think it makes the flow more explicit in this case.